Assets/Scripts/RedRunner/UI/UIScreen/ScoreScreen.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace RedRunner.UI
+{
+    public class ScoreScreen : UIScreen
+    {
+
+        private static ScoreScreen _instance;
+        [SerializeField]
+        private VerticalLayoutGroup layout = default;
+        [SerializeField]
+        private GameObject sliderPrefab;
+        private Dictionary<int, UIScoreBar> scoreBars = new Dictionary<int, UIScoreBar>();
+
+        public void SetVisible(bool visible)
+        {
+            layout.gameObject.SetActive(visible);
+        }
+
+        public static ScoreScreen Instance { get { return _instance; } }
+
+        public void Awake()
+        {
+            if (_instance == null)
+            {
+                _instance = this;
+            }
+        }
+
+        // newPercentage: range [0,1]
+        public void UpdateScore(int id, float newPercentage)
+        {
+            scoreBars[id].SetPercentage(newPercentage);
+        }
+
+        public void CreateScoreBar(int id)
+        {
+            GameObject scoreObject = Instantiate(sliderPrefab, layout.transform);
+            UIScoreBar score = scoreObject.GetComponent<UIScoreBar>();
```

Comment:
I think you can do:
```
UIScoreBar score = Instantiate(sliderPrefab, layout.transform) as UIScoreBar;
```

Reply:
i think for here you have to keep as is, since it's not creating a UIScoreBar, it just has one